sympy/sets/ordinals.py

```diff
@@ -0,0 +1,208 @@
+from sympy.core import Basic, Integer
+import operator
+
+class OmegaPower(Basic):
+    """
+    Represents ordinal exponential and multiplication terms one of the
+    building blocks of the Ordinal class.
+    In OmegaPower(a, b) a represents exponent and b represents multiplicity.
+    """
+    def __new__(cls, a, b):
+        if not all (isinstance(i, (int, Ordinal, Integer)) for i in (a, b)):
+            raise TypeError("arguments must be an integer or an Ordinal instance")
+        return Basic.__new__(cls, a, b)
+
+    @property
+    def exp(self):
+        return self.args[0]
+
+    @property
+    def mult(self):
+        return self.args[1]
+
+    @property
+    def contains_ordinal(self):
+        return any(isinstance(x, Ordinal) for x in self.args)
+
+    def _compare_term(self, other, op):
+        if self.exp == other.exp:
+            return op(self.mult, other.mult)
+        else:
+            return op(self.exp, other.exp)
+
+    def __eq__(self, other):
+        if not isinstance(other, OmegaPower):
+            try:
+                other = Ordinal.convert(other).args[0]
+            except TypeError:
+                return NotImplemented
+        return hash(self) == hash(other)
+
+    def __lt__(self, other):
+        if not isinstance(other, OmegaPower):
+            try:
+                other = Ordinal.convert(other).args[0]
+            except TypeError:
+                return NotImplemented
+        return self._compare_term(other, operator.lt)
+
+    def __hash__(self):
+        return hash((self.exp, self.mult))
+
+class Ordinal(Basic):
+    """
+    Represents ordinals in Cantor normal form.
+
+    Internally, this class is just a list of instances of OmegaPower
+    Examples
+    ========
+    >>> from sympy.sets import Ordinal, Ordinals, OmegaPower
+    >>> Ordinals.w  # doctest: +SKIP
+    w
+    >>> a = Ordinals.w
+    >>> a.is_limit_ordinal
+    True
+    >>> Ordinal([OmegaPower(3,2)])  # doctest: +SKIP
+    {w**3}*2
+    >>> Ordinal([OmegaPower(5,1),OmegaPower(3,2)])  # doctest: +SKIP
+    w**5 + {w**3}*2
+    """
+    def __new__(cls, terms):
```

Comment:
Would it be possible to have a variable number of arguments `*terms` instead of a list?

Reply:
yes, I will update it.